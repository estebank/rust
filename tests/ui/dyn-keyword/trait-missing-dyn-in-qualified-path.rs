//@revisions: edition2018 edition2021
//@[edition2018] edition:2018
//@[edition2021] edition:2021
fn main() {
    let x: u32 = <Default>::default();
    //[edition2021]~^ ERROR expected a type, found a trait
    //[edition2018]~^^ WARNING trait objects without an explicit `dyn` are deprecated
    //[edition2018]~| WARNING this is accepted in the current edition (Rust 2018) but is a hard error in Rust 2021!
    //[edition2018]~| ERROR trait `Default` is not dyn compatible
    //[edition2018]~| ERROR trait `Default` is not dyn compatible
    //[edition2018]~| ERROR the size for values of type `dyn Default` cannot be known at compilation time
    //[edition2018]~| ERROR mismatched types
    //[edition2018]~| ERROR the size for values of type `dyn Default` cannot be known at compilation time
}
