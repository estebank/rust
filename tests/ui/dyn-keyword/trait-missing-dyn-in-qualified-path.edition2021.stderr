error[E0782]: expected a type, found a trait
  --> $DIR/trait-missing-dyn-in-qualified-path.rs:5:19
   |
LL |     let x: u32 = <Default>::default();
   |                   ^^^^^^^
   |
help: you can add the `dyn` keyword if you want a trait object
   |
LL |     let x: u32 = <dyn Default>::default();
   |                   +++

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0782`.
